classes/01class/exercises/c01-git03/roperto/README.md

```diff
@@ -0,0 +1,61 @@
+# c01-git03
+
+## Questions (answer inline)
+
+1. Did you notice any difference in the size of the repo before and after adding the big file?
+```
+Yes, after creating the file it grew 100M as expected. After committing the file it grew around 100M more.
+```
+
+2. What is the reason for this problem?
+```
+That's not exactly a problem, just how git works. It copies the file as an object inside the repository
+so it can be associated with a commit (usually .git directory).
+
+Even if you delete the file from the working directory the "committed" file will still be there.
+
+Even if you delete the commit, most likely the "dangling" commit will still be there (although that can be cleaned).
+```
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+```
+You need to remove the file from every commit and force-push it.
+
+In this case, the commands would be:
+$ git reset @~   # go back before the commit with the big file
+$ git push -f    # we need to force-push as we are rewriting history
+
+You can use the following commands to recover the space locally:
+$ git reflog expire --all --expire-unreachable=now
+$ git gc --aggressive --prune=now
+
+Other alternatives:
+- do an interactive rebase and edit the specific commit.
+- use 'git filter-branch' to remove a file from several commits
+- get a "Big Fuc*ing Gun" that will shoot a green plasma ball obliterate all unwanted stuff on its path DOOM-Style.
```

Comment:
😆 good analogy

There is also `filter-repo`, the recommended alternative to `filter-branch`. 
https://github.com/newren/git-filter-repo